autogpt/config/config.py

```diff
@@ -56,7 +60,24 @@ def __init__(self) -> None:
         self.fast_llm_model = os.getenv("FAST_LLM_MODEL", "gpt-3.5-turbo")
         self.smart_llm_model = os.getenv("SMART_LLM_MODEL", "gpt-4")
         self.fast_token_limit = int(os.getenv("FAST_TOKEN_LIMIT", 4000))
-        self.smart_token_limit = int(os.getenv("SMART_TOKEN_LIMIT", 8000))
+
+        # It's better to set max limits and be explicit for better user experience IMO
+        smart_token_limit = int(os.getenv("SMART_TOKEN_LIMIT", 8000))
+        model_info = OPEN_AI_CHAT_MODELS.get(self.smart_llm_model)
+
+        if model_info is not None:
+            max_limit = model_info.soft_token_limit
+        else:
+            max_limit = smart_token_limit
+
+        if smart_token_limit > max_limit:
+            print(
+                f"Smart LLM model is {self.smart_llm_model}, setting max smart_token_limit to {max_limit}"
+            )
+            smart_token_limit = max_limit
+
+        self.smart_token_limit = smart_token_limit
+
```

Comment:
Should probably set this limit for fast and smart not just smart

Reply:
https://github.com/Significant-Gravitas/Auto-GPT/pull/4128/commits/d1546b7400e34a82ad8aab4cbbbc95129b4cdf3f